### DISCUSSION TITLE
Do not run pipeline after changing kustomization file

### DISCUSSION THREAD

**Discussion by Alissonfsv** [argoproj/argo-cd#20725] [Q&A] [answered]
I am using ArgoCD Image Updater, with the `latest` update strategy. To modify the image tags, I am using the Kustomize method and my repository is GitLab (local). After modifying the kustomize file, it pushes the changes. Depending on the `when:` parameter in gitlab-ci, the pipeline will not be executed automatically. My problem is that my pipeline is always executed, as it is the ideal way for my projects to work. And that is when I came across a looping problem, because imagine the following case:

- The developer makes a change in the code and sends it to the repository;
- The pipeline is executed automatically, thus generating the container image;
- The ArgoCD image updater recognizes that there is a more recent image in the repository, modifies the tag in the kustomize file and sends the modification to GitLab.

With this, you can already imagine how the looping is caused. I found an 'easy' solution to my problem, which would be to run a `git push -o ci.skip` instead of `git push`. Does anyone know if there is a file in the argocd image updater pod where I can change this or something like that?

**Comment by andrii-korotkov-verkada** [comment]
You can create a custom overlay with specifying specific tags for ArgoCD components and update that.

**Reply by Alissonfsv** [reply]
Hello @andrii-korotkov-verkada. Thank you for suggestion. Is there anything in the ArgoCD Image Updater documentation that I could try this with, or is there an example I could use as a reference?

My last attempts were within the application I'm creating in ArgoCD.

**Comment by Alissonfsv** [comment, marked as answer]
Just to conclude this discussion, I return here with the solution I applied, this solution being to add the `-o ci.skip` option to the argocd-image-updater project code and create an image with the changes

Note: All changes were made in the `v0.14.0` branch

`argocd-image-updater/ext/git/writer.go`

```go
// Push pushes local changes to the remote branch. If force is true, will force
// the remote to accept the push.
func (m *nativeGitClient) Push(remote string, branch string, force bool, ciskip bool) error {
	args := []string{"push"}
	if force {
		args = append(args, "-f")
	}
	// Adding ciskip to push
	if ciskip {
		args = append(args, "-o", "ci.skip")
	}
	// End
	args = append(args, remote, branch)
	err := m.runCredentialedCmd(args...)
	if err != nil {
		return fmt.Errorf("could not push %s to %s: %v", branch, remote, err)
	}
	return nil
}
```



`argocd-image-updater/ext/git/client.go`

```go
// Client is a generic git client interface
type Client interface {
	Root() string
	Init() error
	Fetch(revision string) error
	Submodule() error
	Checkout(revision string, submoduleEnabled bool) error
	LsRefs() (*Refs, error)
	LsRemote(revision string) (string, error)
	LsFiles(path string, enableNewGitFileGlobbing bool) ([]string, error)
	LsLargeFiles() ([]string, error)
	CommitSHA() (string, error)
	RevisionMetadata(revision string) (*RevisionMetadata, error)
	VerifyCommitSignature(string) (string, error)
	IsAnnotatedTag(string) bool
	ChangedFiles(revision string, targetRevision string) ([]string, error)
	Commit(pathSpec string, opts *CommitOptions) error
	Branch(sourceBranch string, targetBranch string) error
	// Adding ciskip to the push method
	Push(remote string, branch string, force bool, ciskip bool) error
	// End
	Add(path string) error
	SymRefToBranch(symRef string) (string, error)
	Config(username string, email string) error
}
```



`argocd-image-updater/ext/git/mocks/Client.go`

```go
// Push provides a mock function with given fields: remote, branch, force
// Adding the ci-skip function
func (_m *Client) Push(remote string, branch string, force bool, ciskip bool) error {
	ret := _m.Called(remote, branch, force, ciskip)
// End

	if len(ret) == 0 {
		panic("no return value specified for Push")
	}

	var r0 error
	// Adding the ci-skip
	if rf, ok := ret.Get(0).(func(string, string, bool, bool) error); ok {
		r0 = rf(remote, branch, force, ciskip)
	// End
	} else {
		r0 = ret.Error(0)
	}

	return r0
}
```



`argocd-image-updater/pkg/argocd/git.go`

```go
// Adding ci-skip and setting it to true 
err = gitC.Push("origin", pushBranch, false, true)
	if err != nil {
		return err
	}
// End
```


After that I built the image and defined it in the `Deployment` of `argocd-image-updater`.
